CMakeLists.txt

```diff
@@ -381,9 +374,9 @@ endif()
 
 #-------------------------------------------------------------------------------
 # Compiler Target Options
-# By default, all compiler targets supported by the current platform are
-# enabled. Some compiler targets like CUDA will install external deps as
-# needed at configure time. This can be changed with:
+# By default, all compiler targets supported by the current platform and that don't
+# need a SDK are enabled. Some compiler targets like CUDA will install external
```

Comment:
As HIP/ROCm doesn't need an SDK in the way that CUDA does, we should use slightly different rationale here, and also consider turning off Metal and Vulkan by default too.

Quoting from https://github.com/iree-org/iree/pull/17564#issuecomment-2220760573:

> As someone who has tried at various points to build with everything disabled, I think it would be better for that to be the default in the source. And then we have some documented/checked in way to enable the kitchen sink and that is used consistently in CI and packaging.
> 
> On the cuda/hip front, neither should require the SDK but cuda is currently downloading a ton of stuff to extract the one file it needs (because licensing / none of that stuff is open source). So I think hip is in a better situation here. I don't think either should be enabled in a default build from source config (and neither should other things).

Reply:
Let's start with CUDA and HIP/ROCM here and disable Metal and Vulkan in a follow up.